### PR TITLE
fix: Checkbox align style

### LINKED ISSUE #41564: Changing `controlHeight` breaks vertical alignment of Checkbox component
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-checkbox-sizing-gfov07)

### Steps to reproduce

- Set `controlHeight` of the theme to a larger value than the default.
- Use a `<Checkbox>` component with label text.

### What is expected?

The label of the `checkBox` should stay vertically aligned with the component regardless of `controlHeight`.

### What is actually happening?

The label of the `checkBox` is vertically aligned to the bottom of the component.

| Environment | Info |
| --- | --- |
| antd | 5.3.3 |
| React | 18.2.0 |
| System | Windows 11 Pro Version	10.0.22621 Build 22621 |
| Browser | Google Chrome Version 111.0.5563.147 |

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #41564

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix Check with customize `controlHeight` token not align in center with text.       |
| 🇨🇳 Chinese |   修复 Check 配置 `controlHeight` token 时勾选框与文字不对齐的问题。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at b56db54</samp>

Fixed checkbox vertical alignment and simplified styles. Added a demo to show how to customize checkbox height with `ConfigProvider`.

### <samp>🤖 Generated by Copilot at b56db54</samp>

*  Fix checkbox alignment issue when height is customized ([link](https://github.com/ant-design/ant-design/pull/41566/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL46-R46), [link](https://github.com/ant-design/ant-design/pull/41566/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL74), [link](https://github.com/ant-design/ant-design/pull/41566/files?diff=unified&w=0#diff-c624f3acd6eb939a50b97d0fe9f815bd0dd657eb19797790a8e7b1755ee550efL86-R85))
* Add demo to show how to adjust checkbox height with theme token ([link](https://github.com/ant-design/ant-design/pull/41566/files?diff=unified&w=0#diff-97b787afa8f6dec869cf8e78a17ea86e2923a8f80347fdd3810160e9b1c66588L1-R2), [link](https://github.com/ant-design/ant-design/pull/41566/files?diff=unified&w=0#diff-97b787afa8f6dec869cf8e78a17ea86e2923a8f80347fdd3810160e9b1c66588R46-R55))

### CODE REVIEW

**Review by li-jia-nan — APPROVED** [review]

**Comment by zombieJ** [comment]
Diff 里看到的偏移在涂色后是居中的，过去会偏移 0.2em 使其视觉上对齐。但是考虑到需要符合 token 灵活性，这个 magic 还是要去掉：
截屏2023-03-31 01 06 56

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/39546

这个一起看看？

**Comment by afc163** [comment]
argos 居然有 500+ 个变化

File: components/space/style/index.tsx [hunk 1]
```diff
@@ -42,6 +42,10 @@ const genSpaceStyle: GenerateStyle<SpaceToken> = (token) => {
         '&:empty': {
           display: 'none',
         },
+
+        '*': {
```
Comment (MadCcc) [inline components/space/style/index.tsx:46]: 这个选择器范围有点大，是不是 '> *' 就可以？

Comment (yoyo837) [reply components/space/style/index.tsx:46]: 约束一个名称会不会更好

Comment (zombieJ) [reply components/space/style/index.tsx]: > 这个选择器范围有点大，是不是 '> *' 就可以？

嗯，漏了。

> 约束一个class名称并写到文档会不会更好

Space 里的元素应该都不用考虑 baseline align 的，有的 baseline align 反而会对不齐

**Comment by zombieJ** [comment]
> #39546
> 
> 这个一起看看？

看起来和 #41519 很像，做了一个偏移覆盖。

**Comment by zombieJ** [comment]
Space 改动后，diff 变化太大。revert 掉

**Comment by zombieJ** [comment]
截屏2023-03-31 11 48 00

argos 中发现一例依赖 baseline 对齐逻辑，看来还要换一种实现方式

**Pull request merged**